src/node/chainstatemanager_args.cpp

```diff
@@ -24,7 +24,7 @@
 namespace node {
 util::Result<void> ApplyArgsManOptions(const ArgsManager& args, ChainstateManager::Options& opts)
 {
-    if (auto value{args.GetBoolArg("-checkblockindex")}) opts.check_block_index = *value;
+    if (auto value{args.GetIntArg("-checkblockindex")}) opts.check_block_index = *value;
```

Comment:
In commit "validation: allow to specify frequency for -checkblockindex" (efdc8e3a0ee81b143ee17a578f1e27b1c427a912)

A problem with this change is that it will now interpret bare `-checkblockindex` arguments as `-checkblockindex=0` instead of `-checkblockindex=1`. Could fix this with:

```c++
if (auto value{args.GetIntArg("-checkblockindex")}) {
    // Interpret bare -checkblockindex argument as 1 instead of 0.
   opts.check_block_index = args.GetArg("-checkblockindex")->empty() ? 1 : *value;
}
```

Reply:
Good catch! Took your suggestion. I copied this from the `-checkmempool` parsing code, which will still interpret a bare `-checkmempool` as `-checkmempool=0` (even though that was a boolean arg originally as well). Maybe that should be changed as well.